pkgs/development/libraries/flutter-engine/package.nix

```diff
@@ -0,0 +1,352 @@
+    inherit targetPlatform;
+  };
+
+  git-revision = writeText "git-revision.py" ''
+    #!/usr/bin/env python3
+
+    import sys
+    import subprocess
+    import os
+    import argparse
+
+    def get_repository_version(repository):
+      'Returns the Git HEAD for the supplied repository path as a string.'
+      if not os.path.exists(repository):
+        raise IOError('path does not exist')
+
+      with open(os.path.join(repository, '.git', 'HEAD'), 'r') as head:
+        return head.read().strip()
+
+    def main():
+      parser = argparse.ArgumentParser()
+
+      parser.add_argument(
+        '--repository',
+        action='store',
+        help='Path to the Git repository.',
+        required=True
+      )
+
+      args = parser.parse_args()
+      repository = os.path.abspath(args.repository)
+      version = get_repository_version(repository)
+      print(version.strip())
+
+      return 0
+
+    if __name__ == '__main__':
+      sys.exit(main())
+  '';
+in
+stdenv.mkDerivation {
+  pname = "flutter-engine";
+  inherit version;
+
+  src = runCommand "flutter-engine-source-${version}" {
+    pname = "flutter-engine-source";
+    inherit version;
+
+    inherit (tools) depot_tools;
+
+    nativeBuildInputs = [
+      curl
+      pkg-config
+      git
+      tools.cipd
+      (python3.withPackages (ps: with ps; [
+        httplib2
+        six
+      ]))
+    ];
+
+    gclient = writeText "flutter-engine-${version}.gclient" ''
+      solutions = [{
+        "managed": False,
+        "name": "src/flutter",
+        "url": "${url}",
+      }]
+    '';
+
+    NIX_SSL_CERT_FILE = "${cacert}/etc/ssl/certs/ca-bundle.crt";
+    GIT_SSL_CAINFO = "${cacert}/etc/ssl/certs/ca-bundle.crt";
+    SSL_CERT_FILE = "${cacert}/etc/ssl/certs/ca-bundle.crt";
+    DEPOT_TOOLS_UPDATE = "0";
+    DEPOT_TOOLS_COLLECT_METRICS = "0";
+    PYTHONDONTWRITEBYTECODE = "1";
+
+    outputHashAlgo = "sha256";
+    outputHashMode = "recursive";
+    outputHash = hash.${targetPlatform.system};
+  } ''
+    source ${../../../build-support/fetchgit/deterministic-git}
+    export -f clean_git
+    export -f make_deterministic_repo
+
+    mkdir -p $out
+    cp $gclient $out/.gclient
+    cd $out
+
+    export PATH=$PATH:$depot_tools
+    python3 $depot_tools/gclient.py sync --no-history --shallow --nohooks
+    find $out -name '.git' -exec dirname {} \; | xargs bash -c 'make_deterministic_repo $@' _
+    find $out -path '*/.git/*' ! -name 'HEAD' -prune -exec rm -rf {} \;
+    find $out -name '.git' -exec mkdir {}/logs \;
+    find $out -name '.git' -exec cp {}/HEAD {}/logs/HEAD \;
+
+    echo "${dart.version}" >$out/src/third_party/dart/sdk/version
+
+    python3 src/build/linux/sysroot_scripts/install-sysroot.py --arch=${constants.arch}
+
+    rm -rf $out/.cipd $out/.gclient $out/.gclient_entries $out/.gclient_previous_custom_vars $out/.gclient_previous_sync_commits
+  '';
+
+  toolchain = runCommand "flutter-engine-toolchain-${version}" {
```

Comment:
Explain what is needed? The tool chain derivation? It's because even though Flutter Engine has one, it's broken on nix but we still need to download it or it refuses to configure.

Reply:
I've tried an fhsenv but it wasn't reproducible. It wasn't until last month that I got it reproducible by not using fhsenv.